### DISCUSSION TITLE
Help with product hierarchy design (storyboards.io)

### DISCUSSION THREAD

**Discussion by nenadfspartans** [swellstores/community#334] [Help] [answered]
Hi guys, my client app.storyboards.io would like to integrate swell into their product. We already had some meetings with alex from swell and he helped us about issues we faced. I am asking for help and guidance from community how to solve some issues around products.

We offer creating boards that you can hang on wall with pictures you upload. You have currently 2 options which you can choose: frame and size but in future there could be many more options to build your perfect board.

I did spend some time trying to build something that would suit us but there were always some blockers so any help is appreciated from people with more experience with this service.

Here is our UI that we have currently:

https://i.ibb.co/gwV3hZ6/Screenshot-from-2023-08-08-12-50-25.png
https://i.ibb.co/Mktp0G1/Screenshot-from-2023-08-08-12-50-31.png

Immediately I have tried to create Category named boards and under that placed 3 products named original, canvas, metallic, and on every product I can add variant for 8', 12' and 16'. Problem with this approach is that you cannot add coupon to variant (with swell dashboard, we would like to avoid me creating UI to create those specific coupons) and you cannot specify custom shipping options per variant.

I would also take into account future variants of boards we sell like rectangular, hearth shape, etc...

Looking forward to your suggestions, thanks.

**Comment by ericingram** [comment, marked as answer]
It's true that the Swell dashboard doesn't yet support configuring coupons against variants, but the API does

Seems like something we need to improve cc @aaronlee777

**Reply by aaronlee777** [reply]
I am going to put this down as something for us to look into! 👀
